apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,350 @@
+    const newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  }
+
+  showSchoolInfoForm() {
+    this.setState({showSchoolInfoForm: true});
+  }
+
+  hideSchoolInfoForm() {
+    this.setState({
+      showSchoolInfoForm: false,
+      showSchoolInfoErrors: false,
+    });
+  }
+
+  dismissSchoolInfoForm() {
+    // Clear any state that may have been set
+    this.setState({
+      showSchoolInfoForm: false,
+      country: 'United States',
+      schoolType: '',
+      ncesSchoolId: null,
+      schoolName: '',
+      schoolDisplayName: null,
+      schoolState: '',
+      schoolZip: '',
+      schoolLocation: '',
+      showSchoolInfoErrors: false,
+      showSchoolInfoUnknownError: false,
+    });
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  onSchoolInfoSubmit() {
+    if (this.schoolInfoInputs.isValid()) {
+      $.ajax({
+        url: "/users.json",
+        type: "post",
+        dataType: "json",
+        data: $("#census-school-info-form").serialize()
+      }).done(this.hideSchoolInfoForm.bind(this)).fail(this.updateSchoolInfoError.bind(this));
+    } else {
+      this.setState({
+        showSchoolInfoErrors: true,
+      });
+    }
+  }
+
+  updateSchoolInfoError() {
+    // It isn't clear what could cause an error here since none of the fields are required.
+    this.setState({
+      showSchoolInfoUnknownError: true,
+    });
+  }
+
+  loadSchoolName(schoolId) {
+    if (schoolId && schoolId !== '-1') {
+      $.ajax({
+        url: `/api/v1/schools/${schoolId}`,
+        type: "get",
+      }).done(this.loadSchoolNameSuccess.bind(this)).fail(this.loadSchoolNameError.bind(this));
+    } else {
+      this.setState({
+        schoolDisplayName: ''
+      });
+    }
+  }
+
+  loadSchoolNameSuccess(response) {
+    this.setState({
+      schoolDisplayName: response.name,
+      schoolType: response.school_type,
+    });
+  }
+
+  loadSchoolNameError(error) {
+    this.setState({
+      schoolDisplayName: "your school",
+    });
+  }
+
+  bindSchoolInfoInputs(inputs) {
+    this.schoolInfoInputs = inputs;
+  }
+
+  render() {
+    if (this.state.showSchoolInfoForm) {
+      let schoolId = (this.state.ncesSchoolId !== null) ? this.state.ncesSchoolId : this.props.ncesSchoolId;
+      return (
+        <div style={styles.main}>
+          <form name="census-school-info-form" id="census-school-info-form">
+            <input type="hidden" name="_method" value="patch" />
+            <div style={styles.header}>
+              <h2>Update your school information</h2>
+              {this.state.showSchoolInfoUnknownError && (
+                 <p style={styles.error}>We encountered an error with your submission. Please try again.</p>
+              )}
+            </div>
+            <div style={styles.message}>
+              <SchoolInfoInputs
+                ref={this.bindSchoolInfoInputs}
+                onCountryChange={this.onCountryChange.bind(this)}
```

Comment:
note: you can avoid these explicit bindings by defining the callbacks with arrow functions. See above